github/migrations_source_import_test.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2016 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestMigrationService_StartImport(t *testing.T) {
+	setup()
+	defer teardown()
+
+	input := &Import{
+		VCS:         String("git"),
+		VCSURL:      String("url"),
+		VCSUsername: String("u"),
+		VCSPassword: String("p"),
+	}
+
+	mux.HandleFunc("/repos/o/r/import", func(w http.ResponseWriter, r *http.Request) {
+		v := new(Import)
+		json.NewDecoder(r.Body).Decode(v)
+
+		testMethod(t, r, "PUT")
+		testHeader(t, r, "Accept", mediaTypeImportPreview)
+		if !reflect.DeepEqual(v, input) {
+			t.Errorf("Request body = %+v, want %+v", v, input)
+		}
+
+		w.WriteHeader(http.StatusCreated)
+		fmt.Fprint(w, `{"status":"importing"}`)
+	})
+
+	got, _, err := client.Migrations.StartImport("o", "r", input)
+	if err != nil {
+		t.Errorf("StartImport returned error: %v", err)
+	}
+	want := &Import{Status: String("importing")}
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("StartImport = %+v, want %+v", got, want)
+	}
+}
+
+func TestMigrationService_ImportProgress(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/import", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testHeader(t, r, "Accept", mediaTypeImportPreview)
+		fmt.Fprint(w, `{"status":"complete"}`)
+	})
+
+	got, _, err := client.Migrations.ImportProgress("o", "r")
+	if err != nil {
+		t.Errorf("ImportProgress returned error: %v", err)
+	}
+	want := &Import{Status: String("complete")}
```

Comment:
I typically only do that style if the want value is really simple, like a primitive type.  The struct here does have only a single field, but it's still a bit more complex such that I like leaving it on separate lines, particularly in case we want or need to expand the fields it tests for.